Remarkable-Donkeys/bangazonWebApp
# Mw create product

**The Name of this branch is**
mw-createProduct

* **Please check if the PR fulfills these requirements**


* **What kind of change does this PR introduce?** (Bug Fix, Feature, Document Update...)
Feature


* **What Ticket is associated with this PR**
#1 #3 #14 


* **What is the new behavior?**  (if this is a feature change)
- The create product screen is accessible by clicking the "sell a product" button in the nav bar
- Products are able to be added
- Product photos are able to be uploaded
- Uploaded product images are displayed on the product details page and the product index page

* **List all core functionality**
- Products are added to the product table with all fields complete

* **Other information / Steps to Test**:
1. `git fetch --all`
1. `git checkout mw-createProduct`
1. run the bangazonWepApp
1. click on the "Sell a Product" link in the menu bar
- [x] Confirm that the browser redirects to the Create product page
5. Enter in all information for the product, including uploading an image.
- [x] Confirm that validations are working properly for all fields (except the categories -> I didn't add any functionality for this yet)
 6. Click the 'create' button at the bottom of the page.
- [x] Confirm that the page redirects to the index page.
- [x] Confirm that the product you entered is displayed
- [x] Confirm that a thumbnail of the image you uploaded is displayed for that product.
7. Click the 'details' link for that product. 
- [x] Confirm that the details page is displayed properly with the same image thumbnail. 
8. In Visual Studio, navigate to the wwwroot/images folder
- [x] Confirm that your uploaded photo is located there. 

***At least one person needs to approve pull request before merged to master. Indicate if you think more than one person should test the feature***
